src/esm_runscripts/prepare.py

```diff
@@ -729,3 +730,60 @@ def set_logfile(config):
         "experiment_log_file", log_file_path
     )
     return config
+
+
+def check_config_for_warnings_errors(config):
+
+    triggers = {"error": {"note_function": esm_parser.user_error}}
+
+    # Find conditions to warn (avoid warning more than once)
+    last_jobtype = config["general"].get("last_jobtype", "")
+    isresubmitted = last_jobtype == config["general"]["jobtype"]
+    isinteractive = config["general"].get("isinteractive", "")
+
+    # Only warn if it is an interactive session or while submitted
+    if not isresubmitted or isinteractive:
+        triggers["warning"] = {"note_function": esm_parser.user_note}
+
+    for trigger, trigger_info in triggers.items():
+        warn_error(config, trigger, trigger_info["note_function"])
+
+    return config
+
+def warn_error(config, trigger, note_function):
+
+    if trigger=="warning":
+        sufix_name = f" WARNING"
+    else:
+        sufix_name = f""
+
+    for section, value in config.items():
+        if trigger in value:
+            actions = value[trigger]
+            for action_name, action_info in actions.items():
+                version_condition = action_info.get("esm_tools_version", ">0.0.0")
+                if esm_utilities.check_valid_version(version_condition):
+                    note_function(
+                        f"{action_name}{sufix_name}",
+                        f'Section: ``{section}``\n\n{action_info.get("message", "")}'
+                    )
+
+                    if (
+                        trigger=="warning"
```

Comment:
replace `==` with `is`

Reply:
I disagree with that one. Strings are immutable and it is beeter to use `==` for comparison. `is` operator is used for checking if two objects are identical.

Reply:
This one can be disregarded as well:
```
Python 3.9.5 (default, May 18 2021, 12:31:01)
[Clang 10.0.0 ] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> s = "mystring"
>>> s == "mystring"
True
>>> s is "mystring"
<stdin>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
True
```

I find it easier to ready, but that `SyntaxWarning` will probably pop up in the end user view, and it might be that they at some point make this style of comparisons for strings illegal.

Reply:
Yes, exactly. This is a new feature in Python. When you try to compare immutable types with `is` it tells you that you are doing it wrong. I already fixed the previous ones in the parser.